Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Controllers/MixedRealityControllerVisualizer.cs

```diff
@@ -0,0 +1,285 @@
+            }
+        }
+
+        /// <summary>
+        /// Controller found, create the configured model and position it in the scene
+        /// </summary>
+        /// <param name="eventData"></param>
+        public void OnSourceDetected(SourceStateEventData eventData)
+        {
+            // Capture the respective controller when it's detected.  However, It'll only be rendered if visualization is enabled.
+            if (eventData.Controller != null )
+            {
+                if (eventData.Controller.ControllerHandedness == Handedness.Right)
+                {
+                    rightController = eventData.Controller;
+                }
+                else
+                {
+                    leftController = eventData.Controller;
+                }
+                CreateControllerInstance(eventData.Controller.ControllerHandedness);
+            }
+        }
+
+        /// <summary>
+        /// Controller removed, remove it from the scene
+        /// </summary>
+        /// <param name="eventData"></param>
+        public void OnSourceLost(SourceStateEventData eventData)
+        {
+            // Clean up when a controller is disconnected
+            if (eventData.Controller != null)
+            {
+                DestroyControllerInstance(eventData.Controller.ControllerHandedness);
+            }
+        }
+
+        #endregion SourcePose Event Handlers
+
+        #region Input Handlers
+
+        /// <summary>
+        /// Visualize the pressed button in the controller model, if supported
+        /// </summary>
+        /// <remarks>
+        /// Reserved for future implementation
+        /// </remarks>
+        /// <param name="eventData"></param>
+        public void OnInputDown(InputEventData eventData)
+        {
+            //Visualize button down
+        }
+
+
+        /// <summary>
+        /// Visualize the released button in the controller model, if supported
+        /// </summary>
+        /// <remarks>
+        /// Reserved for future implementation
+        /// </remarks>
+        /// <param name="eventData"></param>
+        public void OnInputUp(InputEventData eventData)
+        {
+            //Visualize button up
+        }
+
+
+        /// <summary>
+        /// Visualize the held trigger in the controller model, if supported
+        /// </summary>
+        /// <remarks>
+        /// Reserved for future implementation
+        /// </remarks>
+        /// <param name="eventData"></param>
+        public void OnInputPressed(InputEventData<float> eventData)
+        {
+            //Visualize single axis controls
+        }
+
+
+        /// <summary>
+        /// Visualize the movement of a dual axis input in the controller model, if supported
+        /// </summary>
+        /// <remarks>
+        /// Reserved for future implementation
+        /// </remarks>
+        /// <param name="eventData"></param>
+        public void OnPositionInputChanged(InputEventData<Vector2> eventData)
```

Comment:
If we're not going to make these methods virtual for inherited classes to override, then let's use the explicit interface declaration.

For Example:
```
void IMixedRealityInputHandler.OnPositionChanged(InputEventData<Vector2> eventData)
```